accelerator/core/mam_core.c

```diff
@@ -37,50 +37,54 @@ int ta_mam_logger_release() {
-                                     trit_t *const msg_id) {
-  retcode_t ret = RC_OK;
-
-  // In TA we only write header on endpoint, since in our architecture we only post Message on signatures owned by
-  // endpoint, but not the signatures owned by channel.
-  ret = mam_api_bundle_write_header_on_channel(api, channel_id, psks, ntru_pks, bundle, msg_id);
+static status_t ta_mam_write_header(mam_api_t *const api, tryte_t const *const channel_id, mam_psk_t_set_t psks,
+                                    mam_ntru_pk_t_set_t ntru_pks, bundle_transactions_t *const bundle,
+                                    trit_t *const msg_id) {
+  retcode_t ret = mam_api_bundle_write_header_on_channel(api, channel_id, psks, ntru_pks, bundle, msg_id);
   if (ret) {
-    ta_log_error("ret: %d, %s\n", ret, error_2_string(ret));
+    ta_log_error("retcode: %d, %s\n", ret, error_2_string(ret));
+    return SC_MAM_FAILED_WRITE_HEADER;
   }
-  return ret;
+  return SC_OK;
 }
 
 /**
- * @brief Writes a packet on a bundle
+ * @brief Writes a packet into a bundle
  *
  * @param api[in] The MAM API object
- * @param bundle[out] The bundle that the packet will be written into
  * @param payload[in] The payload to write
- * @param msg_id[in] The msg_id
+ * @param msg_id[in] The MAM msg_id of this packet
+ * @param bundle[out] The bundle that the packet will be written into
  *
- * @return return code
+ * @return
+ * - SC_OK on success
+ * - non-zero on error
  */
-static retcode_t ta_mam_write_packet(mam_api_t *const api, bundle_transactions_t *const bundle,
-                                     char const *const payload, trit_t const *const msg_id) {
-  retcode_t ret = RC_OK;
+static status_t ta_mam_write_packet(mam_api_t *const api, char const *const payload, trit_t const *const msg_id,
+                                    bundle_transactions_t *const bundle) {
   const bool last_packet = true;
   const mam_msg_checksum_t checksum = MAM_MSG_CHECKSUM_SIG;
   tryte_t *payload_trytes = (tryte_t *)malloc(2 * strlen(payload) * sizeof(tryte_t));
 
   ascii_to_trytes(payload, payload_trytes);
 
-  ret = mam_api_bundle_write_packet(api, msg_id, payload_trytes, strlen(payload) * 2, checksum, last_packet, bundle);
+  retcode_t ret =
+      mam_api_bundle_write_packet(api, msg_id, payload_trytes, strlen(payload) * 2, checksum, last_packet, bundle);
+  if (ret) {
+    ta_log_error("retcode: %d, %s\n", ret, error_2_string(ret));
+    free(payload_trytes);
+    return SC_MAM_FAILED_WRITE;
+  }
 
   free(payload_trytes);
-  return ret;
+  return SC_OK;
 }
 
 /**
  * @brief Find whether the tag of the first transaction of the bundle has already existed on Tangle, which means this
  * current Header has been published.
  *
- * @param tag_array[in,out] A tag array contains all the tags under the given Channel ID
+ * @param tag_array[in/out] A tag array contains all the tags under the given Channel ID
```

Comment:
Why changing `,` to `/`?

Reply:
we use `[in/out]` in most place, so I change it for consistency.

Reply:
But I only find `[in,out]` by searching the doxygen document.